be/src/storage/lake/local_pk_index_manager.cpp

```diff
@@ -86,10 +90,10 @@ void LocalPkIndexManager::gc(UpdateManager* update_manager, DataDir* data_dir, s
         }
         // judge whether tablet should be in the data_dir or not,
         // for data_dir may change if config:storage_path changed.
-        // just remove if not.
+        // just remove dir if not.
         if (StorageEngine::instance()->get_persistent_index_store(id) != data_dir) {
             dir_changed_tablet_ids.push_back(id);
-            if (clear_persistent_index(id).ok()) {
+            if (remove_local_persistent_index_dir(id, data_dir).ok()) {
```

Comment:
please try to add UT to cover the scenario.

Reply:
there is a trick here, i'm working around with it